ros/processor/process_archive.py

```diff
@@ -0,0 +1,66 @@
+import requests
+import logging
+import pydash as _
+from http import HTTPStatus
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+from insights import extract, rule, run, make_metadata
+from insights.parsers.pmlog_summary import PmLogSummary
+from insights.parsers.lscpu import LsCPU
+from insights.parsers.aws_instance_id import AWSInstanceIdDoc
+from insights.core import dr
+from ros.lib.config import INSIGHTS_EXTRACT_LOGLEVEL
+
+
+LOG = logging.getLogger(__name__)
+dr.log.setLevel(INSIGHTS_EXTRACT_LOGLEVEL)
+
+
+@rule(PmLogSummary, LsCPU, AWSInstanceIdDoc)
+def performance_profile(pmlog_summary, lscpu, aws_instance_id):
+    profile = {}
+    performance_metrics = [
+        'mem.physmem',
+        'mem.util.used',
+        'kernel.all.cpu.user',
+        'kernel.all.cpu.sys',
+        'kernel.all.cpu.nice',
+        'kernel.all.cpu.steal',
+        'kernel.all.cpu.idle',
+        'disk.all.total',
+        'mem.util.cached',
+        'mem.util.bufmem',
+        'mem.util.free'
+        ]
+    profile["total_cpus"] = int(lscpu.info.get('CPUs'))
+    profile["instance_type"] = aws_instance_id.get('instanceType')
+    for i in performance_metrics:
+        profile[i] = _.get(pmlog_summary, f'{i}.val', default='None')
+
+    metadata_response = make_metadata()
+    metadata_response.update(profile)
+    return metadata_response
+
+
+def get_performance_profile(report_url):
+    with _download_and_extract_report(report_url) as archive:
+        try:
+            broker = run(performance_profile, root=archive.tmp_dir)
+            result = broker[performance_profile]
+            del result["type"]
```

Comment:
What information this `type` key holds?

Reply:
`{'type': 'metadata', 'total_cpus': 1, 'instance_type': 't2.micro', 'mem.physmem': 825152.0, 'mem.util.used': 663245.405, 'kernel.all.cpu.user': 0.003, 'kernel.all.cpu.sys': 0.001, 'kernel.all.cpu.nice': 0.001, 'kernel.all.cpu.steal': 0.0, 'kernel.all.cpu.idle': 0.994, 'disk.all.total': 2.727, 'mem.util.cached': 351730.563, 'mem.util.bufmem': 335.552, 'mem.util.free': 161906.595}`